src/Npgsql/ConnectorPool.cs

```diff
@@ -310,6 +312,9 @@ internal override void Clear()
     {
         Interlocked.Increment(ref _clearCounter);
 
+        if (Interlocked.CompareExchange(ref _isClearing, 1, 0) == 1)
```

Comment:
I admit I'd use a lock (with Monitor.TryEnter) rather than CompareExchange (just for simplicity), but this is totally fine too.